### PR TITLE
Code cleanup now that all runners support windowed side inputs.

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @charlesccychen

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -98,13 +97,9 @@ def __init__(self,
       self.is_new_dofn = True
 
       # SideInputs
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/common.py]: Can you add a docstring for the constructor of DoFnRunner?  In particular, we should note that the expected type for the side_input argument is a sideinputs.SideInputMap.

File: sdks/python/apache_beam/runners/common.py [hunk 2]
```diff
@@ -98,13 +97,9 @@ def __init__(self,
...
-      self.side_inputs = [side_input
-                          if isinstance(side_input, sideinputs.SideInputMap)
-                          else {global_window: side_input}
-                          for side_input in side_inputs]
+      self.side_inputs = side_inputs
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/common.py]: This is a little confusing now that there is no processing code here; we only need to save the self.side_inputs property when we're using a NewDoFn; can you add a comment to this effect?

**Comment by charlesccychen** [comment]
Thanks! Looks good modulo comments for clarity.

File: sdks/python/apache_beam/runners/common.py [hunk 3]
```diff
@@ -71,6 +70,21 @@ def __init__(self,
                # Preferred alternative to context
                # TODO(robertwb): Remove once all runners are updated.
                state=None):
+    """Initializes a DoFnRunner.
+
+    Args:
+      fn: user DoFn to invoke
+      args: positional side input arguments (static and placeholder), if any
+      kwargs: keyword side input arguments (static and placeholder), if any
+      side_inputs: list of sideinput.SideInputMaps for deferred side inputs
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/common.py]: nit: this should be "sideinputs.SideInputMaps"; period at end of each description.

**Comment by charlesccychen** [comment]
Thanks! LGTM (modulo nit).

**Pull request closed without merging**